src/main/java/org/spongepowered/api/command/CommandSource.java

```diff
@@ -34,7 +35,7 @@
  * <p>Examples of potential implementations include players, the server console,
  * Rcon clients, web-based clients, command blocks, and so on.</p>
  */
-public interface CommandSource extends Subject {
+public interface CommandSource extends ContextViewer, Subject {
```

Comment:
This might be a little interesting to handle, I'm not 100% sure that this is going to be easily implementable with command block sources trying to execute a command according to another player? Or is this intended for other use cases?